### ISSUE TITLE
onProviderChange not detect gps change correctly

### ISSUE THREAD

**Issue by miguelflores1993** [transistorsoft/flutter_background_geolocation#771]
## Your Environment
* Plugin version:  4.6.3
* Platform: Android
* OS version: 
* Device manufacturer / model: Xiaomi MI 11T
* Flutter info (`flutter doctor`):

## Expected Behavior
When I disable gps, onProviderChange should return gps false, and if enabled it should return true.
But it is not working correctly, sometimes it returns well, other times it returns badly.

## Context
```
bg.BackgroundGeolocation.onProviderChange((bg.ProviderChangeEvent event) {
      switch (event.status) {
        case bg.ProviderChangeEvent.AUTHORIZATION_STATUS_DENIED:
          break;
        case bg.ProviderChangeEvent.AUTHORIZATION_STATUS_ALWAYS:
          if (event.gps) {
            bg.BackgroundGeolocation.getCurrentPosition(persist: false, samples: 1).then((location) {
              add(MyLocationUpdateEvent(location: LatLng(location.coords.latitude, location.coords.longitude)));
            });
          } else {
            add(MyLocationUpdateEvent());
          }
          break;
        case bg.ProviderChangeEvent.AUTHORIZATION_STATUS_WHEN_IN_USE:
          if (event.gps) {
            bg.BackgroundGeolocation.getCurrentPosition(persist: false, samples: 1).then((location) {
              add(MyLocationUpdateEvent());
            });
          }
          break;
      }
    });
```

**Comment by christocracy** [comment]
> When I disable gps,

How exactly are you disabling “gps” (you probably mean “disable location”.

show me a screenshot of what you are disabling.

**Comment by miguelflores1993** [comment]
yes exactly when I disable the location.

`for example I have a log when I deactivate the location and the gps parameter returns true, and that is wrong.`

```
D/TSLocationManager(21791): [c.t.locationmanager.util.c h]
D/TSLocationManager(21791):   ℹ️  LocationAuthorization: Permission granted
I/TSLocationManager(21791): [c.t.l.service.HeartbeatService c]
I/TSLocationManager(21791):   🔴  Stop heartbeat
D/TSLocationManager(21791): [c.t.locationmanager.util.c h]
D/TSLocationManager(21791):   ℹ️  LocationAuthorization: Permission granted
I/TSLocationManager(21791): [c.t.l.service.TrackingService a]
I/TSLocationManager(21791):   🔵  setPace: false → false
I/TSLocationManager(21791): [c.t.l.a.BackgroundGeolocation$a1 onReceive]
I/TSLocationManager(21791): ╔═════════════════════════════════════════════
I/TSLocationManager(21791): ║ Location-provider change: true
I/TSLocationManager(21791): ╠═════════════════════════════════════════════
I/TSLocationManager(21791): ╟─ GPS: true
I/TSLocationManager(21791): ╟─ Network: false
I/TSLocationManager(21791): [c.t.l.s.ActivityRecognitionService b]
I/TSLocationManager(21791):   🎾  Start motion-activity updates
D/TSLocationManager(21791): [c.t.locationmanager.util.c h]
```

**Comment by christocracy** [comment]
Show me the screenshot of what you’re disabling.

**Comment by miguelflores1993** [comment]
![image](https://user-images.githubusercontent.com/67565934/174390785-b91c5e13-3295-4e02-ae04-6fb4960005cc.png)

**Comment by christocracy** [comment]
Confirmed.

- Fixed in branch `release-4.7.0`
- Test it out by installing from Git:

```yaml
dependencies:
  flutter_background_geolocation:
    git:
      url: https://github.com/transistorsoft/flutter_background_geolocation.git
      ref: release-4.7.0
```

**Comment by BecoStopclub** [comment]
Hello @christocracy , I have a problem running _4.6.3_ and _4.7.0_ on Android.
Location is not tracked, not even the notification appears.
Version 4.6.2 works perfectly.

A log that appears to me:

```‼️  Failed to find SingleLocationRequest in START action; requestId: 7; Intent: Intent { act=start cmp=xxxx/com.transistorsoft.locationmanager.service.LocationRequestService (has extras) }```

**Comment by BecoStopclub** [comment]
Forget, I just figured out, I was doing something like that:

```dart
final state = await bg.BackgroundGeolocation.setConfig(...);
if (!state.enabled) {
    await bg.BackgroundGeolocation.setOdometer(0);
    final newState = await bg.BackgroundGeolocation.start();
    return newState.enabled;
}
```

The `Failed to find SingleLocationRequest` log was caused by:
```dart
await bg.BackgroundGeolocation.setOdometer(0);
```

After remove it everything works fine, it was working in version _4.6.2_

**Comment by christocracy** [comment]
I will investigate the setOdometer issue.

**Comment by christocracy** [comment]
It’s not a serious error.  It’s not going to prevent the plugin working as expected.

**Comment by christocracy** [comment]
>‼️ Failed to find SingleLocationRequest in START action;

Fixed and pushed to branch `release-4.7.0`.  Do a `flutter pub get`

```
‼️ Failed to find SingleLocationRequest in START action;
```

When you see `Build 406` from the plugin logs in `adb logcat`, you have the fix 
```
╔═════════════════════════════════════════════
║ TSLocationManager version: 3.2.0 (406)
╠═════════════════════════════════════════════
```

**Comment by BecoStopclub** [comment]
It's working, thanks 💯
